extensions/sql-migration/src/constants/helper.ts

```diff
@@ -187,8 +187,10 @@ export function canCutoverMigration(migration: DatabaseMigration | undefined): b
 	const status = getMigrationStatus(migration);
 	return hasMigrationOperationId(migration)
 		&& isOnlineMigration(migration)
-		&& (status === loc.MigrationState.ReadyForCutover || status === loc.MigrationState.InProgress)		// TODO: InProgress condition can be eventually deprecated
-		&& isFullBackupRestored(migration);
+		&& (status === loc.MigrationState.ReadyForCutover || status === loc.MigrationState.InProgress)
+		&& isFullBackupRestored(migration)
+		// if MI migration, must have no restore blocking reason
+		&& !(getMigrationTargetType(migration) === loc.SQL_MANAGED_INSTANCE && (migration?.properties.migrationStatusWarnings?.restoreBlockingReason ?? '').length > 0);
```

Comment:
nit: maybe add a method 'hadRestoreBlockingReason'

Reply:
thanks Brian, good catch. Added a new helper method `hasRestoreBlockingReason` and updated other references to use this